src/Workspaces/Core/Portable/Execution/Serialization/Checksum.cs

```diff
@@ -73,8 +75,8 @@ public override int GetHashCode()
                     hash = Hash.Combine((int)_checkSum[i], hash);
                 }
 
-                _lazyHash = hash;
-            }
+                _lazyHash = new StrongBox<int>(hash);
+            };
```

Comment:
The first may or may not be an issue.  It depends on how LocalSlotManager compiler team will have to say for sure.  The latter definitely looks like an issue :)

Don't compound it though.  Let's just be non-racy from the beginning, instead of getting into a place where we're spending tons of man hours months from now tracking down a hard to repro failure.